packages/forklift-console-plugin/src/modules/Providers/utils/components/CertificateUpload/CertificateUpload.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { FC } from 'react';
+import { useModal } from 'src/modules/Providers/modals';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import { Button, FileUpload, FileUploadProps, Flex, FlexItem } from '@patternfly/react-core';
+
+import { FetchCertificateModal } from './FetchCertificateModal';
+
+export interface CertificateUploadProps {
+  id?: string;
+  url: string;
```

Comment:
`url` is optional

if not available, fall back to regular `FileUpload`

Reply:
Actually this is already a regular `FileUpload`:)

Reply:
in what way is this the regular `FileUpload`, does `FileUpload` has a required `url` field,  what am i missing ?

Reply:
the component is actually (pseudo-code):
```
CertificateUpload = () => <FileUpload> <Button/> </FileUpload>
```
how you would like to change it?

Reply:
> how you would like to change it?

when you use `FileUpload`:

`<FileUpload .... >`

when you use `CertificateUpload`:

`<CertificateUpload ....  url="some-url" >`
or
 `<CertificateUpload .... >`
 
 so a user can safely use `CertificateUpload` instead of  `FileUpload` as a "drop in replacment"
 
 and the `Button` from your example
 
` CertificateUpload = () => <FileUpload> <Button/> </FileUpload>
`

will be optional, and render only if the `url` prop is not `undefined`